examples/lorawan/main.c

```diff
@@ -148,6 +149,16 @@ int main(void)
     puts("LoRaWAN Class A low-power application");
     puts("=====================================");
 
+    /*
+     * Enable deep sleep power mode (e.g. STOP mode on STM32) which
+     * in general provides RAM retention after wake-up.
+     */
+#if IS_USED(MODULE_PM_LAYERED)
+    for (unsigned i = 1; i < PM_NUM_MODES; ++i) {
+        pm_unblock(i);
+    }
+#endif
```

Comment:
I don't get this. Why would one decrement a ref counter with no coordination?

It triggers the assertion in `pm.c` on non-STM32 boards right away, because not every ref counter is higher than 1. But this is bound to break any periph driver than explicitly requested a certain clock.

We should rather try to adapt the `periph_uart` implementation to not keep a clock required after TX is completed when no RX callback is set.